### DISCUSSION TITLE
cdk deploy causing many s3 requests?

### DISCUSSION THREAD

**Discussion by Ju66ernaut** [aws/aws-cdk#28139] [Q&A]
I am making a back end project in aws cdk. I have 8 different stacks. I broke them up mostly by service so I have an api gateway stack, a lambda stack, vpc stack, iam stack, etc. I used the getting started instructions in the aws docs to start a cdk app. An s3 bucket (versioning enabled) prefixed with "cdk-" and then a hash and then the region was created and I see this is where my zip and json files are going when I do `cdk deploy`.

Today I got an alert saying I have almost reached my 2000 request limit for the free tier. The app I am making does have document upload functionality but I have used it maybe 10 times. Looking through the s3 bucket metrics for that cdk bucket I see when I run `cdk deploy` 22 requests are made. 2 GET, 2 PUT and 18 list requests?

My assumption is cdk deploy gets the latest 2 versions of the zip and json files, does some comparison with the 2 local versions and uploads the 2 new versions. But I can't figure out where the other list requests are coming from and why so many? I can't find anything about this in the documentation. 

**tl;dr - doing cdk deploy is causing 22 requests each time and while I'm in the development phase of my app (making a lot of changes) I am almost hitting the request cap of the free tier by just deploying code. Does anyone have an idea what this could be and how I can remedy it?**
